modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -364,7 +367,11 @@ private void declareMessageSource(HasSourceDeclarer declarer, Class<? extends So
 
         sourceDeclarers.put(sourceType, source);
         declareMetadataKeyId(sourceType, source);
-        declareSingleParameters(getParameterFields(sourceType), source, MuleExtensionAnnotationParser::parseMetadataAnnotations);
+        declareSingleParameters(getParameterFields(sourceType)
+                                        .stream()
+                                        .filter(field -> !isMultiLevelMetadataKeyId(field, field.getType(), typeLoader))
```

Comment:
I understand what you said, from my point of view source parameters should be treated in the same way as operations, but due to the coupled logic iis impossible to be done with out refactoring all parameter parsing logic